### DISCUSSION TITLE
Why `store-only` on this line?

### DISCUSSION THREAD

**Discussion by frederikhors** [jycouet/kitql#91] [Q&A] [answered]
I do not understand why `store-only` on the line:

https://github.com/frederikhors/demo-kitql-sveltekit/blob/17de09c1e84277b4b016af126934c7c333d3cd80/src/routes/continents/%5Bcode%5D/index.svelte#L19

I need the urql's `cache-and-network` behavior and if I use `cache-and-store` it doesn't work! It patch Antartica if I choose Europe.

Maybe it's me but I cannot create a demo of what I need. Can you help me understand?

**Comment by frederikhors** [comment]
![image](https://user-images.githubusercontent.com/41120635/160025814-930b2c1a-e1c2-4f02-9f5c-82ebfe0b52d2.png)

With `cache-and-store`.

**Comment by jycouet** [comment, marked as answer]
You **NEED** a lot of things mister @frederikhors !
Quite complicated to follow you in all your jumps right / left / middle.
You open a discussion here on another repo... would be easier there no?!

`patch` cannot do `cache-and-network` as it's something local to the site. Patch will never do a request.
`query` will do a request (it's the line just under: `21`)
My idea is that you `patch` manually your data with what you know, and the `query` will fill the store automatically just after with the info coming from the network.

Now, you show me a screenshot with undefined... It's a bit complicated to know where you are... In a client side navigation? First hit? You changed my code? 

If it patched `Antartica if I choose Europe`, that mean that you manually didn't patch the right thing no?!
On my side the demo was fully working.

How should we proceed?!

**Reply by jycouet** [reply]
Let me know @frederikhors what are you doing differently, maybe I don't have your use case in mind?

https://user-images.githubusercontent.com/5312607/160028290-3390887a-0235-4fb5-a497-e7bde7da3a67.mov

**Reply by frederikhors** [reply]
First of all, sorry if the tone seemed complaining or even angry: ABSOLUTELY NOT! I'm just a little tired. 😄

> You open a discussion here on another repo... would be easier there no?!

Sorry, I thought I was on that one.

I thought better and tried and I also made some changes and **everything works as I would like**.

You are a genius, really. ❤️

I have a lot of confusion about patch types. This is only because lack of docs and because I still have to practice.

**Reply by frederikhors** [reply]
Your video appeared after I posted the message! Fake real-time from Github!

**Reply by frederikhors** [reply]
The change I needed was this: `{#if $KQL_AllCountriesOfContinent.isFetching && !$KQL_AllCountriesOfContinent.data}` here: https://github.com/frederikhors/demo-kitql-sveltekit/commit/f4da85a118bd6ef51134d7abfde714b01f037072.

**Reply by jycouet** [reply]
Thx for the reply.
Good that you understand a bit better how thing are in my mind. (Not in doc yet, but I'm working on it 🙃)
